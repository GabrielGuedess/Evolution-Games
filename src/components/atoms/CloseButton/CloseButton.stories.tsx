import * as Popover from '@radix-ui/react-popover';
import { Story, Meta } from '@storybook/react/types-6-0';

import { CloseButton } from './CloseButton';

export default {
  title: 'Atoms/CloseButton',
  component: CloseButton,
} as Meta;

export const Default: Story = () => (
  <Popover.Root>
    <CloseButton />
  </Popover.Root>
);
