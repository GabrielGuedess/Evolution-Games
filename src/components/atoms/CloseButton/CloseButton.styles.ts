import * as Popover from '@radix-ui/react-popover';
import media from 'styled-media-query';

import styled, { css } from 'styled-components';

export const Close = styled(Popover.Close)`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    padding: 0.5rem;
    transition: color 0.3s linear;
    cursor: pointer;
    display: flex;
    gap: 0.5rem;
    font-size: 1.2rem;
    align-items: center;
    position: absolute;
    top: 0.5rem;
    left: 2.1rem;

    svg {
      cursor: pointer;
    }

    ${media.greaterThan('medium')`
      left: 0.5rem;
    `}
  `}
`;
