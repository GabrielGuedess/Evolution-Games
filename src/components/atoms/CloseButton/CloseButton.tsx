import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react';

import { X } from 'phosphor-react';

import * as S from './CloseButton.styles';

type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>;

export type CloseButtonProps = {
  side?: 'left' | 'right';
  color?: 'primary' | 'white';
  as?: React.ElementType;
} & ButtonTypes;

export const CloseButton = ({
  side = 'left',
  color = 'primary',
  ...props
}: CloseButtonProps) => (
  <S.Close side={side} color={color} {...props} aria-label="Close Action">
    <X size={20} />
  </S.Close>
);
