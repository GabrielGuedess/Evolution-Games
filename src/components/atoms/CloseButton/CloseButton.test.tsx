import * as Popover from '@radix-ui/react-popover';
import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { CloseButton } from './CloseButton';

describe('<CloseButton />', () => {
  it('should render correctly', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Popover.Root>
        <CloseButton />
      </Popover.Root>,
    );

    // Assert
    expect(screen.getByLabelText(/Close Action/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
