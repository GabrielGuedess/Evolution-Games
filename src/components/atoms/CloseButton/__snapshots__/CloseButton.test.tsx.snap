// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<CloseButton /> should render correctly 1'] = `
.c0 {
  color: #855aff;
  padding: 0.5rem;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  font-size: 1.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 0.5rem;
  left: 2.1rem;
}

.c0 svg {
  cursor: pointer;
}

@media (min-width:768px) {
  .c0 {
    left: 0.5rem;
  }
}

<button
  aria-label="Close Action"
  class="c0"
  type="button"
>
  <svg
    fill="currentColor"
    height="20"
    viewBox="0 0 256 256"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="256"
      width="256"
    />
    <line
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="200"
      x2="56"
      y1="56"
      y2="200"
    />
    <line
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="200"
      x2="56"
      y1="200"
      y2="56"
    />
  </svg>
</button>
`;
